### ISSUE TITLE
PromiseConnection.ping() doesn't reject

### ISSUE THREAD

**Issue by mdierolf** [sidorares/node-mysql2#812]
I noticed while reviewing the code in promise.js that ping() doesn't have the possibility to ever reject, which doesn't seem correct:
```js
PromiseConnection.prototype.ping = function() {
  var c = this.connection;
  return new this.Promise(function(resolve, reject) {
    c.ping(resolve);
  });
};
```

Shouldn't it be:
```js
PromiseConnection.prototype.ping = function() {
  const c = this.connection;
  const localErr = new Error();
  return new this.Promise(function(resolve, reject) {
    const done = makeDoneCb(resolve, reject, localErr);
    c.ping(done);
  });
};
```

**Comment by sidorares** [comment]
yes, you are right. At the protocol level ping always respond with OK packet, the error could probably only happen if connection was closed before or during ping command ( which is most common use case for ping - check if it's still alive )

**Comment by sidorares** [comment]
work on fix in #813
